### PR TITLE
Modified styles and assets

### PR DESCRIPTION
**Pull request by sanketbansal**
#### Changes proposed in this pull request:

- Added assets from current EvalAI version.
- Added styles required for modifying pages.
- Added and modified the fonts.
- All added styles are palced in styles-port folder to support previous styles also.

- Link to live demo: http://pr-154-evalai.surge.sh

### CODE REVIEW

**Comment by sanketbansal** [comment]
@lunayach @Shekharrajak  I have made this seperate PR to reduce the file changes and repeated commits in other dependent PRs.

**Comment by lunayach** [comment]
Screenshot from 2019-06-23 23-23-49

@sanketbansal this is the screenshot that I took for the current UI. You may want to resolve some issues.

**Review by lunayach — CHANGES_REQUESTED** [review]
@sanketbansal try to resolve the issues for this PR first so that we could review the other pending PR's.

**Comment by sanketbansal** [comment]
@lunayach I have just added the styles in this PR. Not integrated them with the components that i will be doing in other PR for each component separately.

**Comment by sanketbansal** [comment]
Also, I have kept the separate folder for these styles so that old styles can be applied.

**Comment by lunayach** [comment]
Okay @sanketbansal, But could you refer all the pushed PR's which are depending on this (after attaching a screenshot in each of them), so that we could see if modified styles/assets are making sense.

**Comment by sanketbansal** [comment]
#150  , #158 , #159 , #140 , #139 , #149 , #143 , #141 
All these PRs depends on these styles directly.

**Comment by sanketbansal** [comment]
@lunayach  please review.

File: src/styles/styles-port/modules/notification.scss
Comment (lunayach) [inline src/styles/styles-port/modules/notification.scss]: Things like this could be removed.

**Comment by lunayach** [comment]
@Shekharrajak @RishabhJain2018 Should we merge this PR first and we could see UI changes later in the above mentioned PR’s specifically?

**Review by lunayach — APPROVED** [review]

**Comment by Sanji515** [comment]
@sanketbansal @lunayach @RishabhJain2018 @galipremsagar @Shekharrajak 

I think `vendor-port` css file is from the EvalAI project which is generated using gulp, which contains almost all the css code of EvalAI in one file and I think we should not copy exactly the same file.

**Comment by lunayach** [comment]
Oh yes, it’s an auto-generated file. @sanketbansal please remove this. Thanks, @Sanji515 for spotting this.

**Comment by Sanji515** [comment]
And also I would suggest that as Angular 7 has differnet components and each components has it's `scss` file then we don't need to create these extra files and import them in `main.scss`. Even we don't need that `main.scss` file

**Review by lunayach — COMMENTED** [review]
Ideally we should have component wise .scss. But since @sanketbansal has already done this way, we might need to discuss this.

**Comment by sanketbansal** [comment]
@lunayach 

> Oh yes, it’s an auto-generated file. @sanketbansal please remove this. Thanks, @Sanji515 for spotting this.

Actually, vender-port.css files contain only the CSS related to third-party libraries i.e animate.css, angular material and toaster, trix and materialize.
So, I thought if we can use their raw CSS  instead of using these libraries as a whole. Then it will be great
Also, we can remove unused CSS from that. once complete implementing different components.

**Comment by lunayach** [comment]
@sanketbansal Good thought, but I don't think we should ever make changes in the third-party libraries unless we have to. Otherwise, future developers/maintainers would have to scratch their heads for these trivial things :p.

And you need not worry about the unused stuff and all, Angular build when builds after optimisation, anyways makes files of very small sizes.

**Comment by sanketbansal** [comment]
> @sanketbansal Good thought, but I don't think we should ever make changes in the third-party libraries unless we have to. Otherwise, future developers/maintainers would have to scratch their heads for these trivial things :p.

No, I am not making any changes in these libraries as I said I am just using their CSS properties as it is.
But can remove some unused properties if it is required.
Also if UI of the component lacks something then I add styles for the component specifically. Just like we do in angular 7.

**Comment by Sanji515** [comment]
I think the libraries they build, they use css for their things like rich text editor, angular material and in angular 7 if we want to use similar things then there are packages available. So instead of writing css we should directly install those packages along with their css.

**Comment by sanketbansal** [comment]
yeah @Sanji515  we can do that, But I was talking about minor bugs such as margin or padding issue or color issue.

**Comment by Sanji515** [comment]
> But I was talking about minor bugs such as margin or padding issue or color issue.

@sanketbansal if you are talking to do these minor changes inside the CSS of those packages, then of course you can simply override them, let me know if I didn't get you

**Comment by sanketbansal** [comment]
NO, I am talking about the changes in styles of component specifically inside **component.scss** file.

**Review by Shekharrajak — APPROVED** [review]
I believe you have made changes according to EvalAI properly.

**Comment by Sanji515** [comment]
@Shekharrajak But I think it will contain the two css files for one component one inside the component and another outside which is included inside the `main.scss` which is not good as structure wise and also we don't need to include the `vendor-port.css` file

**Comment by lunayach** [comment]
@RishabhJain2018 just take a look at the comments thread above before merging.

**Comment by RishabhJain2018** [comment]
@sanketbansal Why are we adding `vendor-port.scss` file? I don't see it's need.

**Comment by RishabhJain2018** [comment]
I'm merging it for now to unblock you but we might need to clean this up in future.

**Comment by sanketbansal** [comment]
> @sanketbansal Why are we adding `vendor-port.scss` file?

@RishabhJain2018 is explained that here  https://github.com/Cloud-CV/EvalAI-ngx/pull/154#issuecomment-505857371

**Comment by RishabhJain2018** [comment]
> So, I thought if we can use their raw CSS instead of using these libraries as a whole. 

Can you please list pros/cons of this?

**Comment by RishabhJain2018** [comment]
> And also I would suggest that as Angular 7 has differnet components and each components has it's scss file then we don't need to create these extra files and import them in main.scss. Even we don't need that main.scss file

Moreover, I'd prefer it to do this way which is a more cleaner approach.

**Comment by sanketbansal** [comment]
I kept those styles separate in the **styles-port** folder so that these styles do not get mixed with the previous ones as @Sanji515  was also working on the project. 
I was looking to change them according to the preferred format once the important PRs get merged.

**Comment by sanketbansal** [comment]
> > And also I would suggest that as Angular 7 has differnet components and each components has it's scss file then we don't need to create these extra files and import them in main.scss. Even we don't need that main.scss file
> 
> Moreover, I'd prefer it to do this way which is a more cleaner approach.

I have already moved the styles to their respective components. In other PRs.

**Comment by lunayach** [comment]
> I have already moved the styles to their respective components. In other PRs.

@sanketbansal does this mean that you have moved the likes of `challenge-create.scss`, `challenge.scss` etc. to their respective component folders?

**Comment by Shekharrajak** [comment]
Actually we were trying to unblock you but it seems if we merge this PR then it will remain same and it would be hard to refactor it. 

Every component must contain its own CSS but here we can see all things are in one folder (only common CSS should be outside the component folder).

**Comment by Shekharrajak** [comment]
I think @Sanji515 can help to categorize the css and put it into the corresponding component folder. If there are multiple css files for one component then it should be in `compnent_name/css` (I just want to see good codebase structure).

**Comment by sanketbansal** [comment]
@lunayach I have moved the challenge.scss and challenge-create file from the style-port folder into their respective components(#143, #150).

**Comment by lunayach** [comment]
> @lunayach I have moved the challenge.scss and challenge-create file from the style-port folder into their respective components(#143, #150).

@sanketbansal  Okay, so what's wrong in doing the same in this PR itself? If not done, then again they would come under new changes in those PRs. That was the very purpose of this PR, I guess.

**Comment by Shekharrajak** [comment]
Let's do one thing:

@Sanji515 can you clone this branch and try to put CSS files in the corresponding components and push new commits?

**Comment by sanketbansal** [comment]
I have done some changes. Please review

**Comment by lunayach** [comment]
@sanketbansal are the changes done?

**Comment by sanketbansal** [comment]
@lunayach working on it. will commit those shortly.

**Comment by sanketbansal** [comment]
@lunayach I have made those changes please review.
Also if any other changes required then please comment.

File: src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts
Comment (lunayach) [inline src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts]: Why is this changed in "modified styles.." PR?

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html
Comment (lunayach) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html]: Same for this.

File: src/styles/main.scss
Comment (lunayach) [inline src/styles/main.scss]: How is this file different from `base.scss`? Doesn't `base.scss` serve a similar purpose?
@Sanji515 may also help in the discussion here.

File: src/styles/main.scss
Comment (lunayach) [inline src/styles/main.scss]: How is this file different from `base.scss`? Doesn't `base.scss` serve a similar purpose?
@Sanji515 may also help in the discussion here.

**Review by lunayach — CHANGES_REQUESTED** [review]
1. Changes other than the 'modified styles and assets' are included.
2. Some other comments.

**Comment by lunayach** [comment]
@sanketbansal Can you let us know which feature/dependency is (or going to be) dependent on `vendor-port.css`? Because it's an auto-generated file. (@Sanji515 may also join this discussion here.)

**Comment by Sanji515** [comment]
@lunayach @sanketbansal what I think that this PR even can be closed and the styles which are needed for all other PR's can be added in that only, cause the code I've written so far in the PR's is generalised in a way that we only need to write css if anything new there in the UI (for that we'll copy those css from this PR) otherwise we can use the already existing css which are in `base.scss`.

And also as this PR is not blocking to me and @sanketbansal for now so I think we can come to it later when all the current task are completed.

**Comment by lunayach** [comment]
Yes, @Sanji515 I agree that we can add styles for the components on which we are working on. but then this PR (#140) has **137** file changes!  
Maybe for now we could include only the static files in this PR.
(@sanketbansal )

**Comment by Sanji515** [comment]
Then in `Modified homepage` PR, we can delete all the styles inside `styles-port`. Also we can delete the images which are included inside `images-port` as I see someone of them are already included inside `assets/images` in the master.

**Comment by lunayach** [comment]
Yes! Does this sound good to you @sanketbansal?

**Comment by sanketbansal** [comment]
> Then in `Modified homepage` PR, we can delete all the styles inside `styles-port`. Also we can delete the images which are included inside `images-port` as I see someone of them are already included inside `assets/images` in the master.

was not this the whole purpose from the beginning to merge this PR first to reduce the no of file changes and then remove unnecessary files in the corresponding PRs.

**Comment by lunayach** [comment]
Yes, @sanketbansal that was the reason. If you could keep static files in this PR and resolve my other comments for the files in this PR, then we are good to go.

**Comment by sanketbansal** [comment]
So I am removing all the styles from this PR. Okay

**Comment by lunayach** [comment]
@sanketbansal  It would be better if you could reply inline to the comments raised in the Files as I had some doubts there as well.

File: src/styles/main.scss
Comment (sanketbansal) [reply src/styles/main.scss]: Yeah, so should I copy the styles from main.scss to base.scss?

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html
Comment (sanketbansal) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html]: I will remove that

File: src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts
Comment (sanketbansal) [reply src/app/components/challenge/challengeoverview/challengeoverview.component.spec.ts]: will remove that

**Comment by sanketbansal** [comment]
> @sanketbansal Can you let us know which feature/dependency is (or going to be) dependent on `vendor-port.css`? Because it's an auto-generated file. (@Sanji515 may also join this discussion here.)

Auth pages use some styles from vendor-port also dashboard component use styles from vendor port. I am removing the vendor-port.scss anyway.

File: src/styles/main.scss
Comment (lunayach) [reply src/styles/main.scss]: Yes! just copy the relevant styles that are required now to the `base.scss`.

**Comment by sanketbansal** [comment]
@lunayach please review

File: src/styles/base.scss
Comment (lunayach) [inline src/styles/base.scss]: I am not sure if all these styles are (or are going to be) used. In future PR's, @sanketbansal @Sanji515 try to remove redundant and unused styles from here.

Comment (sanketbansal) [reply src/styles/base.scss]: yeah, I think that we can do in other dependent PRs

**Review by lunayach — APPROVED** [review]
@RishabhJain2018 I think we can merge this now.

File: src/styles/base.scss
Comment (Sanji515) [reply src/styles/base.scss]: @lunayach yeah of course some changes are new but most of them are already on the master, this is the reason I'm saying to close this PR and do the css changes in their respective PR so that it would be easy to track what should not be included in the PR. These fonts can also be included in their respective PR's

Comment (lunayach) [reply src/styles/base.scss]: I don't think that the `base.scss` is having changes for the already pushed PR's. That should not be the case anyways. Is it correct, @sanketbansal?

Comment (sanketbansal) [reply src/styles/base.scss]: so, @Sanji515 can you tell me the changes which have already been made in the master?
Then I will remove them

Comment (sanketbansal) [reply src/styles/base.scss]: @lunayach I have tried to added changes related to my PRs.
Also, U and @Sanji515  can run the PR on local machine and check if its hindering with already made changes.

Comment (lunayach) [reply src/styles/base.scss]: Ah, let's not go this way then. Use updated master's `base.scss` for all your future PR's. 
Let's also remove `base.scss` from this PR. (That is, only static files for now.)
That way, I guess it would be easy to keep a track of all the changes done for a particular functionality.

**Comment by Sanji515** [comment]
@sanketbansal Also I would suggest to remove all the images from `image-port` as you have ported all the images of current version in which some of them are not being used and also it contains some images which are already on the `master`

File: src/styles/base.scss
Comment (sanketbansal) [reply src/styles/base.scss]: I have checked there is no duplicate styles present in base.scss file. Also no style is hindering with the master branch. Also these added styles are required by the other components.

**Review by lunayach — CHANGES_REQUESTED** [review]
@sanketbansal also move the images from `images-port` to `images` folder.

**Comment by lunayach** [comment]
@sanketbansal I still see the `images-port` folder. Did you move all the images?

**Comment by Sanji515** [comment]
@sanketbansal The organizations images that you put inside the `organizations` folder are already there inside `images/partners` folder. So I think you should check for the duplicate images there and at other places also.

File: src/styles/base.scss
Comment (lunayach) [reply src/styles/base.scss]: (IMO) if `base.scss` is also having styles for already pushed PR's, then we should not include it in this PR. We can easily afford one extra file change in the form of `base.scss` for future PR's.

Also, can you confirm for @Sanji515's comment:

> @sanketbansal The organizations' images that you put inside the organizations' folder are already there inside images/partners folder. So I think you should check for the duplicate images there and at other places also.

So, that means only static file changes for this PR. @sanketbansal I understand that you had to make lot's of changes for this (and other related) PR's, but to keep the project structure clean, I guess, it's worth it.

**Comment by sanketbansal** [comment]
There were only 7 duplicate image assets present which I have removed.

> @sanketbansal The organizations images that you put inside the `organizations` folder are already there inside `images/partners` folder. So I think you should check for the duplicate images there and at other places also.

yeah, I know that I am using organization folder in my PRs and did not removed partner folder so that in case you will be using that.
Anyways, I have removed that organization folder.

File: src/styles/base.scss
Comment (sanketbansal) [reply src/styles/base.scss]: That's why I was keeping a separate file of my changes. And was looking to add that into base.scss after all PRs got merged.

Comment (lunayach) [reply src/styles/base.scss]: But can we add those changes PR by PR as we proceed (in `base.scss`)?

Comment (sanketbansal) [reply src/styles/base.scss]: Any issue with variable.scss file?
Also, should I make changes in base.scss in other PRs or not?

Comment (lunayach) [reply src/styles/base.scss]: `Variable.scss` is fine.
Sure, you can make changes in `base.scss` for the functionality that the other PR introduces.

Comment (sanketbansal) [reply src/styles/base.scss]: > But can we add those changes PR by PR as we proceed (in `base.scss`)?

For that I have to check each and every style of the HTML in other PR and then have to search whether that is already present or not which is going to be a tedious task.

Comment (lunayach) [reply src/styles/base.scss]: I think to avoid possible duplicities and adding those things only which we actually need, it should be done. We could also take @RishabhJain2018's take on this.

Comment (RishabhJain2018) [reply src/styles/base.scss]: I agree with @lunayach.

**Comment by sanketbansal** [comment]
removed styles from base.scss. please check

**Review by lunayach — APPROVED** [review]
@RishabhJain2018 this seems good to me now. This will not overlap/duplicate the already pushed (or future) changes.

**Comment by sanketbansal** [comment]
@RishabhJain2018 I am waiting for this PR to be merged

**Comment by sanketbansal** [comment]
@lunayach is it okay if I add a common.scss file to include styles which is common for different components.

**Comment by lunayach** [comment]
@sanketbansal if the styles are common, they'll anyways be reflected for only the first PR that'll we merge. I don't think we need another `common.scss`.

**Comment by sanketbansal** [comment]
> @sanketbansal if the styles are common, they'll anyways be reflected for only the first PR that'll we merge. I don't think we need another `common.scss`.

okay

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**
